pkg/util/util.go

```diff
@@ -836,18 +836,22 @@ func GetAndExtractZip(zipURL string, destination string) error {
 		return errors.Errorf("Invalid Zip URL: %s . Should either be prefixed with file://, http:// or https://", zipURL)
 	}
 
-	_, err := Unzip(pathToZip, destination)
+	filenames, err := Unzip(pathToZip, destination, pathToUnzip)
 	if err != nil {
 		return err
 	}
 
+	if len(filenames) == 0 {
+		return errors.Errorf("No files were unzipped, ensure that the project repo is not empty or that sparseCheckoutDir has a valid path")
+	}
+
 	return nil
 }
 
 // Unzip will decompress a zip archive, moving all files and folders
 // within the zip file (parameter 1) to an output directory (parameter 2).
 // Source: https://golangcode.com/unzip-files-in-go/
```

Comment:
We should update the function description to explain the usage of `pathToUnzip` argument, also correct the description as we don't always `moving all files and folders within the zip file`